context/ThemeProvider.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { getCookie } from 'cookies-next';
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+} from 'react';
+
+import { IThemeContext, ModeType } from '@/types';
+
+const ThemeContext = createContext<IThemeContext | undefined>(
+  undefined
+);
+
+export const ThemeProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [mode, setMode] = useState<ModeType>('light');
+
+  const themeCookie = getCookie('theme');
+
+  function handleThemeChange() {
+    if (themeCookie === 'dark') {
+      setMode('light');
+      document.documentElement.classList.remove('dark');
+    } else {
+      setMode('dark');
+      document.documentElement.classList.add('dark');
+    }
```

Comment:
```suggestion
    if (themeCookie === 'dark') {
      setMode('dark');
      document.documentElement.classList.add('dark');
    } else {
      setMode('light');
      document.documentElement.classList.remove('dark');
    }
```

The logic for this is a little backwards. This isn't like the themeswitch where we need to `toggle` the theme. Here we're responding to the theme change from the switch and simply setting our document classes/mode to reflect that state switch. Currently if you look at your cookies you have "dark" when light mode renders.